bindings/distrdf/python/DistRDF/Ranges.py

```diff
@@ -407,3 +486,182 @@ def get_clustered_range_from_percs(percrange: TreeRangePerc) -> Tuple[Optional[T
                           globalstart, globalend, percrange.friendinfo)
 
     return treerange, entries_in_trees
+
+def get_percentage_ranges_sample(treenames: List[str], filenames: List[str], npartitions: int,
+                          sampleMap: Dict[str, SerializableRSample], 
+                          friendinfo: Optional[ROOT.Internal.TreeUtils.RFriendInfo],
+                          exec_id: ExecutionIdentifier) -> List[TreeRangeSamplePerc]:
```

Comment:
I would try to think about whether this function and the next could be simplified, or removed. I could be missing something, but from what I can see right now, they are fairly similar if not practically the same of their counterparts without the `_sample` part of the name, except for the fact that we need to pass the `sample` down to the new `TreeRangeSamplePerc` or `TreeRangeSample`.

As an idea, I can imagine we could add `samples: Optional[List[SerializableRSample]]` to the already existing `TreeRangePerc` and `TreeRange`, then optionally inject the sample infomation in `get_percentage_ranges`.